### ISSUE TITLE
关于CV_CUDA性能和精确度的测试疑问[QUESTION]

### ISSUE THREAD

**Issue by sunxirui310** [CVCUDA/CV-CUDA#35]
**What is your question?**
感谢分享这么棒的成果。写了一个简单的cpu、gpu 图像resize的测试脚本。测试后产生一些疑惑。
`import torch`
`import torchnvjpeg`
`import numpy as np`
`import cvcuda`
`import cv2`
`if __name__=="__main__":`
`    batch_size = 1`
`    max_cpu_threads = 8`
`    device_id = 0`
`    max_image_size = 3840 * 2160 * 3`
`    target_img_height = 224`
`    target_img_width = 128`
`    decoder = torchnvjpeg.Decoder(0, 0, True, device_id, batch_size, max_cpu_threads, max_image_size, torch.cuda.current_stream(device_id))`
`    `
`    #read img files`
`    image_path = "images/catbig.jpg"`
`    data = open(image_path, 'rb').read()`
`    data_list = [data for _ in range(batch_size)]`
`    `
`    #decode img using gpu`
`    t0_gpu=cv2.getTickCount()`
`    image_tensor_list = decoder.batch_decode(data_list)`  
`    image_tensors = torch.stack(image_tensor_list)  `
`    cvcuda_input_tensor = cvcuda.as_tensor(image_tensors, "NHWC")`
`    t1_gpu=cv2.getTickCount()`
`    `
`    # Resize to the input network dimensions`
`    cvcuda_resize_tensor = cvcuda.resize(`
`                cvcuda_input_tensor,`
`                (`
`                    batch_size,`
`                    target_img_height,`
`                    target_img_width,`
`                    3,`
`                ),`
`                cvcuda.Interp.LINEAR,`
`            )`
`    torch_preprocessed_tensor = torch.as_tensor(`
`        cvcuda_resize_tensor.cuda(), device="cuda"`
`    )`
`    t2_gpu=cv2.getTickCount()`
`    im_resize_gpu=torch_preprocessed_tensor.cpu().numpy()[0,:,:,::-1]#rgb2bgr`
`    `
`    #cpu`
`    t0_cpu=cv2.getTickCount()`
`    im_cpu=cv2.imdecode(np.frombuffer(data,np.uint8),1)`
`    t1_cpu=cv2.getTickCount()`
`    im_resize_cpu=cv2.resize(im_cpu,(target_img_width,target_img_height))#opencv default bilinear`
`    t2_cpu=cv2.getTickCount()`
`    `
`    print('gpu: read and resize an image')`
`    print(f'    --decoding: {(t1_gpu-t0_gpu)/cv2.getTickFrequency()*1000:.4f}ms')`
`    print(f'    --reszie: {(t2_gpu-t1_gpu)/cv2.getTickFrequency()*1000:.4f}ms')`
`    print('cpu: read and resize an image')`
`    print(f'    --decoding: {(t1_cpu-t0_cpu)/cv2.getTickFrequency()*1000:.4f}ms')`
`    print(f'    --reszie: {(t2_cpu-t1_cpu)/cv2.getTickFrequency()*1000:.4f}ms')`
`    same=np.isclose(im_resize_cpu,im_resize_gpu,0.001)`
`    print(f'input image size is {im_cpu.shape[1]}x{im_cpu.shape[0]}.\n\`
`after resize：{im_resize_cpu.shape[1]}x{im_resize_cpu.shape[0]}. \n\`
`CPU&GPU same element percent:{np.sum(same)/np.size(same)*100:.6f}%')`
`    cv2.imwrite("CPU.jpg",im_resize_cpu)`
`    cv2.imwrite("GPU.jpg",im_resize_gpu)`

结果是
![image](https://user-images.githubusercontent.com/19179730/212849446-674faf5f-9962-46b2-a8ec-8f1eba698dbb.png)
机器环境Win10+WSL，1080Ti，i7 7700K。

几个疑问如下
1. gpu的resize操作时间长，是代码写法有问题吗？用于推理的话如何加速？
2. opencv的resize和CV_CUDA的resize结果差异还是挺大的。没有跟opnecv结果对齐是硬件问题吗？

再此求助答疑解惑，谢谢

**Comment by sunxirui310** [comment]
![catbig](https://user-images.githubusercontent.com/19179730/212854177-6423f7cf-8433-4115-bda0-54230daf8004.jpg)

**Comment by mkhadatare** [comment]
Hi [sunxirui310](https://github.com/sunxirui310),

Can you test for batch of images like 32, 64 or 128 then your performance is higher. 

Also CV-CUDA interpolation method is LINEAR whereas OpenCV using Bilinear, can you use same method on both side.

**Comment by milesp-nvidia** [comment]
Can you attempt this test again with the Beta release of CV-CUDA.

**Comment by zhkuo24** [comment]
From what I've tested, it looks like GPU computing goes through a warm-up process. So, you should start timing from the second computation onwards.

**Comment by zhkuo24** [comment]
From what I've tested, it looks like GPU computing goes through a warm-up process. So, you should start timing from the second computation onwards. @sunxirui310

**Comment by bhaefnerNV** [comment]
Closing due to inactivity. Please don't hesitate to reopen if more questions arise.
Thank you!
